src/plugins/dashboard/public/dashboard_api/get_dashboard_state.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { pick } from 'lodash';
+import moment, { Moment } from 'moment';
+import { RefreshInterval } from '@kbn/data-plugin/public';
+
+import { convertPanelMapToPanelsArray, extractReferences, generateNewPanelIds } from '../../common';
+import type { DashboardAttributes } from '../../server';
+
+import { convertDashboardVersionToNumber } from '../services/dashboard_content_management_service/lib/dashboard_versioning';
+import { GetDashboardStateProps } from '../services/dashboard_content_management_service/types';
+import {
+  dataService,
+  embeddableService,
+  savedObjectsTaggingService,
+} from '../services/kibana_services';
+import { LATEST_DASHBOARD_CONTAINER_VERSION } from '../dashboard_container';
+
+export const convertTimeToUTCString = (time?: string | Moment): undefined | string => {
+  if (moment(time).isValid()) {
+    return moment(time).utc().format('YYYY-MM-DDTHH:mm:ss.SSS[Z]');
+  } else {
+    // If it's not a valid moment date, then it should be a string representing a relative time
+    // like 'now' or 'now-15m'.
+    return time as string;
+  }
+};
+
+export const getDashboardState = ({
```

Comment:
Can this be renamed to `getSerializedState` (and rename file to `get_serialized_state`)? That will avoid ambiguity of what state this is returning.